### PR TITLE
Submission Review (grading) with multiple metrics

### PR DESCRIPTION
**Pull request by eladeyal-intel**
see frab#518

### CODE REVIEW

**Comment by eladeyal-intel** [comment]
Added test for editing an event score's and checking that the average is calculated correctly.

**Comment by erdgeist** [comment]
There's a lot going on in this PR that doesn't have to do with review metrics, like introduction of the assistant role and the concept of stakeholders. This makes it quite hard to review the review metrics part.

Since I am not sure how this is supposed to work, anyway, I focussed on anything BUT the review metrics and this looks okay.

**Comment by eladeyal-intel** [comment]
@erdgeist - I updated #518 to show better what "review metrics" are supposed to do.

**Comment by eladeyal-intel** [comment]
- Changes the settings tab to be called "Reviewing" and not "Review Metrics". Updated #518 accordingly.
- Simplified ReviewMetric::safe_name so now it has a default value for names only included non-latin characters.

File: app/models/event.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class Event < ApplicationRecord
   include ActionView::Helpers::TextHelper
   include EventState
   include HasEventConflicts
-
+
```
Comment (erdgeist) [inline app/models/event.rb]: whut space?

File: app/views/event_ratings/show.html.haml [hunk 2]
```diff
@@ -75,6 +75,17 @@
         %fieldset.inputs
           = f.input :rating, as: :rating
           = f.input :comment, input_html: {rows: 3}
+          = f.simple_fields_for :review_scores do |ff|
+            = ff.input :score,
+              as: :radio_buttons,
+              collection: [[t('events_module.not_applicable'), '0'], ['1', '1'], ['2', '2'], ['3', '3'], ['4', '4'], ['5', '5']],
```
Comment (erdgeist) [inline app/views/event_ratings/show.html.haml]: Does the zero value mean that you can undo a rating?

Comment (eladeyal-intel) [reply app/views/event_ratings/show.html.haml]: Yes. The value 0 is displayed as N/A when rating; and as a "blank" when looking at the rating summary. It does not taken account when calculating average review.

image

**Comment by erdgeist** [comment]
I understand most of the PR now and see little changes that have the potential not be corrected easily if they contain mistakes I have not spotted.

I wonder how good the interface will actually work in the event reviews pane, but these may also be patched up later.

So from me that would be a go. Maybe @manno wants to comment, if not I'll merge after 24h.

**Comment by eladeyal-intel** [comment]
https://github.com/frab/frab/pull/614 is this pull + a modification to add_fake_data to create fake reviews, if you want to experiment with the GUI

**Pull request merged**